oci/oci.go

```diff
@@ -124,12 +124,26 @@ func (r *Runtime) CreateContainer(c *Container) error {
 	childPipe.Close()
 
 	// Wait to get container pid from conmon
-	// TODO(mrunalp): Add a timeout here
-	var si *syncInfo
-	if err := json.NewDecoder(parentPipe).Decode(&si); err != nil {
-		return fmt.Errorf("reading pid from init pipe: %v", err)
+	s := make(chan *syncInfo, 1)
+	e := make(chan error, 1)
+	go func() {
+		var si *syncInfo
+		if err := json.NewDecoder(parentPipe).Decode(&si); err != nil {
+			e <- fmt.Errorf("reading pid from init pipe: %v", err)
+			return
+		}
+		s <- si
+	}()
+
+	t := time.NewTimer(time.Second * 3)
+	select {
+	case si := <-s:
+		logrus.Infof("Received container pid: %v", si.Pid)
+	case err = <-e:
+		return err
+	case <-t.C:
```

Comment:
No, the timeout should be set here but you're right that it would be a bunch of code changes that aren't necessarily worth the effort. The reason why the timeout might want to be configurable is if/when we start adding more setup and hook code to conmon (the hooks might take longer than 3 seconds to start up, but we're going to fail the creation because of the limit).

Also @mrunalp does erroring here correctly clean up the partially-created container?